relentless/potential/potential.py

```diff
@@ -586,19 +705,29 @@ def regularize(self, u, f, trim=True):
         if self.fcut is not None:
             flags = np.abs(np.flip(f)) >= self.fcut
             cut = len(f)-1 - np.argmax(flags)
-            u -= u[cut]
             if cut < len(f)-1:
+                if self.shift:
+                    u -= u[cut]
                 u[(cut+1):] = 0.
                 f[(cut+1):] = 0.
+            else:
+                warnings.warn('Last tabulated force exceeds fcut, rmax may be too small.', UserWarning)
+                if self.shift:
+                    u -= u[-1]
+        elif self.shift:
+            u -= u[-1]
 
         # trim off trailing zeros
         r = self.r.copy()
-        if trim:
-            flags = np.abs(np.flip(f)) > 0
-            cut = len(f) - np.argmax(flags)
-            if cut < len(f)-1:
+        flags = np.abs(np.flip(f)) > 0
+        cut = len(f)-1 - np.argmax(flags)
```

Comment:
I think this might need this to go back to
```suggestion
        cut = len(f) - np.argmax(flags)
```
So that we always include the last point where the force and energy are both zero.

Right now, I think this will throw out all of the points where the force is zero, right? By retaining one point where the force is zero, we ensure that the tabulated force and energy both interpolate to zero at the end point.

The rest should then work as is.

@zeesherman what do you think?